app/controllers/invites_controller.rb

```diff
@@ -40,12 +41,16 @@ def show
 
   def perform_accept_invitation
     params.require(:id)
-    params.permit(:username, :name, :password, :timezone, user_custom_fields: {})
+    params.permit(:email, :username, :name, :password, :timezone, user_custom_fields: {})
     invite = Invite.find_by(invite_key: params[:id])
 
-    if invite.present?
+    if invite.present? && !invite.expired?
       begin
-        user = invite.redeem(username: params[:username], name: params[:name], password: params[:password], user_custom_fields: params[:user_custom_fields], ip_address: request.remote_ip)
+        user = if invite.is_invite_link?
+          invite.redeem_invite_link(email: params[:email], username: params[:username], name: params[:name], password: params[:password], user_custom_fields: params[:user_custom_fields], ip_address: request.remote_ip)
+        else
+          invite.redeem(username: params[:username], name: params[:name], password: params[:password], user_custom_fields: params[:user_custom_fields], ip_address: request.remote_ip)
```

Comment:
We were already testing for everything except `link_valid?`, added check for that too. Note that `!redeemed?` and `!destroyed?` is being checked for in `show` method which is a precursor to `perform_accept_invitation`.

Reply:
I feel like the flow of logic here is very complex because the conditions for `Invite#redeem_invite_link` and `Invite#redeem` is duplicated in the model and in the controller. Perhaps we can check that `user` is `nil` and return an error message instead?

Reply:
> Note that !redeemed? and !destroyed? is being checked for in show method which is a precursor to perform_accept_invitation.

The method can be called directly via API as well so it doesn't always go through the `show` method.

Reply:
> Perhaps we can check that user is nil and return an error message instead?

Agreed. Updated the code to send response based on user record presence.